src/main/java/uk/gov/hmcts/reform/authorisation/healthcheck/ServiceAuthHealthIndicator.java

```diff
@@ -16,20 +21,55 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(ServiceAuthHealthIndicator.class);
 
     private final ServiceAuthorisationHealthApi serviceAuthorisationHealthApi;
+    private final String secret;
+    private final String microService;
+    private final ServiceAuthorisationApi serviceAuthorisationApi;
 
     @Autowired
-    public ServiceAuthHealthIndicator(final ServiceAuthorisationHealthApi serviceAuthorisationHealthApi) {
+    public ServiceAuthHealthIndicator(
+            ServiceAuthorisationHealthApi serviceAuthorisationHealthApi,
+            @Value("${idam.s2s-auth.totp_secret}") String secret,
+            @Value("${idam.s2s-auth.microservice}") String microService,
+            ServiceAuthorisationApi serviceAuthorisationApi
+    ) {
         this.serviceAuthorisationHealthApi = serviceAuthorisationHealthApi;
+        this.secret = secret;
+        this.microService = microService;
+        this.serviceAuthorisationApi = serviceAuthorisationApi;
     }
 
     @Override
     public Health health() {
+        return check();
+    }
+
+    private Health check() {
         try {
             InternalHealth internalHealth = this.serviceAuthorisationHealthApi.health();
-            return new Health.Builder(internalHealth.getStatus()).build();
+            if (!internalHealth.getStatus().getCode().equalsIgnoreCase(Status.UP.getCode())) {
+                return new Health.Builder(internalHealth.getStatus()).build();
+            } else {
+                String token = generateToken();
+                String serviceName = this.serviceAuthorisationApi.getServiceName(token);
```

Comment:
this library is used by services that are not registered in s2s (they only validate incoming service) thus this check cannot be performed.

Reply:
Was not aware of this.

Reply:
@rkondratowicz can you give an example of service that just uses `getServiceName` to validate token?

Reply:
draft-store and send-letter

Reply:
Thanks @rkondratowicz.